src/main/java/com/google/api/codegen/transformer/ModelToJavaTypeTable.java

```diff
@@ -0,0 +1,255 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.LanguageUtil;
+import com.google.api.codegen.java.JavaTypeTable;
+import com.google.api.codegen.util.TypeAlias;
+import com.google.api.tools.framework.model.ProtoElement;
+import com.google.api.tools.framework.model.ProtoFile;
+import com.google.api.tools.framework.model.TypeRef;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+import com.google.protobuf.DescriptorProtos.FieldDescriptorProto.Type;
+
+import java.io.File;
+import java.util.List;
+
+public class ModelToJavaTypeTable implements ModelTypeTable {
+  private JavaTypeTable javaTypeTable;
```

Comment:
The fact that this holds mutable state is not obvious, nor is it obvious that the state is a list of import statements. Also, the decomposition into `ModelToJavaTypeTable` and `JavaTypeTable` is not obvious.

Similar thoughts apply to the PHP code.